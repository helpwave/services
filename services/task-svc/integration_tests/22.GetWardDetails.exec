#!/bin/bash

echo "Exists"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "0f13614c-9043-4af7-a439-cc569ba22639"}' \
	localhost:3001 \
	proto.services.task_svc.v1.WardService/GetWardDetails

echo "Exists, no beds"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "626e8c70-dc93-44fb-a9d5-0b828a0a0c4b"}' \
	localhost:3001 \
	proto.services.task_svc.v1.WardService/GetWardDetails

echo "Does not exists:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "a7d943f6-dae1-48c9-b947-8523c214f297"}' \
	localhost:3001 \
	proto.services.task_svc.v1.WardService/GetWardDetails 2>&1 || echo ""
