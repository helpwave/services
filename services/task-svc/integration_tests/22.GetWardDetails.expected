Exists
{
  "id": "0f13614c-9043-4af7-a439-cc569ba22639",
  "name": "Ohio",
  "rooms": [
    {
      "id": "b79dc72f-8359-4058-bd7a-e4b7019a5ec3",
      "name": "Sales",
      "beds": [
        {
          "id": "cb92bad3-68b2-49c1-bfac-494ff18b57f3",
          "name": "Aquamarine"
        },
        {
          "id": "31d2cb94-c6fd-4d98-acd4-128c31d51693",
          "name": "Khaki"
        },
        {
          "id": "03d95f53-06f1-4163-b9c2-4aa03d1bcc5e",
          "name": "Green"
        },
        {
          "id": "edef8f4f-fa11-471c-9340-c667ae6ecf09",
          "name": "Indigo"
        },
        {
          "id": "9a2824d8-cd31-4c5f-82df-1c6bf1d443c6",
          "name": "Pink"
        }
      ]
    },
    {
      "id": "2a3b9111-1f78-4d36-8318-1c69c1e5ad51",
      "name": "Whatever else"
    }
  ],
  "taskTemplates": [
    {
      "id": "aff58dea-8665-48c0-9ddc-e58727d4c8b4",
      "name": "Removal of Synth Sub from L Fibula, Perc Endo Approach",
      "subtasks": [
        {
          "id": "ffee7b8b-5663-458e-a5d2-8f9e27112ee8",
          "name": "Extraction of Left Pleura, Open Approach, Diagnostic"
        }
      ]
    },
    {
      "id": "1abafcba-be0f-4d00-84a2-2ee79e44efca",
      "name": "Insertion of Radioactive Element into R Knee, Open Approach",
      "subtasks": [
        {
          "id": "f43cc18a-9c5b-4c6b-a63d-ff2a149d47ee",
          "name": "HDR Brachytherapy of L Breast using Cesium 137"
        },
        {
          "id": "cfcb37fc-fe5b-48bf-b1db-c0543a3d8596",
          "name": "Masking Patterns Assessment using Sound Field / Booth"
        },
        {
          "id": "c9560437-d537-4932-ba4c-3b799894fd03",
          "name": "Fusion of Left Elbow Joint, Percutaneous Approach"
        },
        {
          "id": "88c3ba49-fa57-4ae0-a07b-8542528a0a0f",
          "name": "Fluoroscopy of Left Pulmonary Vein using Other Contrast"
        }
      ]
    },
    {
      "id": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd",
      "name": "Reattachment of Right Foot, Open Approach",
      "subtasks": [
        {
          "id": "86617de9-4012-4ee3-9ecb-1fc0c9e7ab85",
          "name": "Reposition R Metacarpocarp Jt w Ext Fix, Perc Endo"
        },
        {
          "id": "5a712e12-c2a0-472e-827a-0e754a396378",
          "name": "Change Traction Apparatus on Right Lower Extremity"
        },
        {
          "id": "a77d20cb-1954-4e4f-baed-a5e28c0b620e",
          "name": "Destruction of Left Trunk Tendon, Perc Endo Approach"
        },
        {
          "id": "972830e0-28f8-4e28-8187-e63e11514915"
        },
        {
          "id": "c6e6bde8-ca25-4a8f-80b5-cc6697d8f523",
          "name": "Removal of Radioactive Element from L Lung, Extern Approach"
        }
      ]
    },
    {
      "id": "93ccbea0-26a6-4331-a8df-e6ffb040e949",
      "name": "Excision of Tonsils, Percutaneous Approach",
      "subtasks": [
        {
          "id": "e35bbe38-3b0d-4108-a90d-c8b0cbe7b1e5",
          "name": "Repair Radial Nerve, Open Approach"
        },
        {
          "id": "1543d756-306c-4019-baaf-74f72672ba58",
          "name": "Insert of Infusion Dev into L Radial Art, Perc Endo Approach"
        }
      ]
    },
    {
      "id": "aa62dc26-b81f-47b1-9542-1f1ee3ebafb1",
      "subtasks": [
        {
          "id": "b61c5618-e926-4171-9cd7-7f6c9050bac8",
          "name": "Division of Lumbar Vertebra, Open Approach"
        }
      ]
    },
    {
      "id": "5f634e01-1e7e-422d-9322-695f186e908d",
      "name": "Destruction of Bilateral Fallopian Tubes, Perc Endo Approach",
      "subtasks": [
        {
          "id": "7ea1ede4-6326-4de2-8ea0-aaf0320fd666",
          "name": "Drain R Metacarpophal Jt w Drain Dev, Perc Endo"
        }
      ]
    },
    {
      "id": "ee863790-1faa-4610-b845-39ba878a59fe",
      "name": "Muscle Perform Trmt Neuro Up Back/UE w Prosthesis",
      "subtasks": [
        {
          "id": "0840c5e4-835e-493f-a15d-ef0e03aefb0b",
          "name": "Dilate R Com Iliac Art w 2 Intralum Dev, Perc Endo"
        },
        {
          "id": "0956f068-73d0-4035-a015-a38594b28ef4"
        }
      ]
    },
    {
      "id": "a544bd78-8c45-4034-8c79-7621f36816ee",
      "name": "Repair Uterine Supporting Structure, Open Approach",
      "subtasks": [
        {
          "id": "78d0233f-184c-4154-a26e-4aa17fcdf78d",
          "name": "Insert of Radioact Elem into R Low Arm, Perc Endo Approach"
        }
      ]
    },
    {
      "id": "0e4b502e-fc08-4c24-a819-da6da17e6ce8",
      "name": "Introduction of Other Hormone into Periph Art, Perc Approach",
      "subtasks": [
        {
          "id": "2e9cecf9-776d-48d6-bc1d-1935bd5f4e86"
        }
      ]
    },
    {
      "id": "9ee21d4a-65cc-431a-8ded-32790ddd8be1",
      "subtasks": [
        {
          "id": "8a8e3f56-cfb2-4fe7-afdc-9a3feed13c9a",
          "name": "Drainage of Liver with Drainage Device, Perc Endo Approach"
        }
      ]
    },
    {
      "id": "378dc856-b4e0-47b1-908c-01b4ec974a6d",
      "name": "Restrict R Com Iliac Vein w Intralum Dev, Perc Endo",
      "subtasks": [
        {
          "id": "c8f04797-24c7-4f23-b53d-f56991e5b32b",
          "name": "Reposition Left Shoulder Muscle, Open Approach"
        }
      ]
    },
    {
      "id": "fa7a8940-7a2c-466e-a0d5-0b4ee5c2208a",
      "name": "Insertion of Monitor Dev into Tracheobronc Tree, Via Opening",
      "subtasks": [
        {
          "id": "722661aa-5071-4322-ad30-cb114d92ea64",
          "name": "Restriction of Azygos Vein, Percutaneous Endoscopic Approach"
        }
      ]
    },
    {
      "id": "835041e7-1989-4583-9ca4-cd98e91b24e4",
      "name": "Reposition Coccyx, External Approach",
      "subtasks": [
        {
          "id": "6ffa9e28-6266-4db5-a409-8c918e298177",
          "name": "Dilate R Verteb Art, Bifurc, w 4+ Intralum Dev, Perc Endo"
        },
        {
          "id": "1760bdd8-978c-4dcd-9786-78af8c9078e6",
          "name": "Drainage of R Hip Bursa/Lig with Drain Dev, Open Approach"
        }
      ]
    },
    {
      "id": "12dedb6e-de87-471d-b812-44fbfaa9e73f",
      "subtasks": [
        {
          "id": "fe6b8a5e-bffc-4cc8-bdfb-883bd6392ac2",
          "name": "Bypass R Ventricle to L Pulm Art w Nonaut Sub, Perc Endo"
        }
      ]
    },
    {
      "id": "dd3f7c98-2b61-4b61-922a-d730bfb53ae7",
      "name": "Bypass 1 Cor Art from R Int Mammary w Synth Sub, Perc Endo",
      "subtasks": [
        {
          "id": "2767aa0d-a4dc-4cda-8d4f-030b272b6ede"
        }
      ]
    },
    {
      "id": "c314dc38-9c90-431b-8fc1-1ae3b4174a27",
      "name": "Excision of Left Internal Mammary Artery, Perc Approach"
    },
    {
      "id": "52c9430f-614f-428a-b327-56ae07ebbe5c",
      "name": "Supplement R Up Arm Tendon with Synth Sub, Open Approach"
    },
    {
      "id": "47d91496-1351-48be-9f62-09cd819cf9fc",
      "name": "Removal of Nonaut Sub from Bladder, Perc Approach"
    },
    {
      "id": "40ee9af4-9037-4d56-bf1b-81c16b3ab871",
      "name": "Pharmacotherapy for Substance Abuse Treatment, Naloxone"
    },
    {
      "id": "e89950be-63f5-4122-90d8-f5696b81b933",
      "name": "Removal of Synthetic Substitute from Cerv Jt, Open Approach"
    },
    {
      "id": "5e72a85e-89b0-4056-9d0c-51432193c108",
      "name": "Restrict of R Post Tib Art with Extralum Dev, Open Approach"
    },
    {
      "id": "c2e7b56b-9302-41b4-842d-31970ecf4a12",
      "name": "Supplement R Radial Art with Autol Sub, Perc Endo Approach"
    },
    {
      "id": "9661984d-0ee2-4f61-8302-00eeb7b23a37",
      "name": "Motor Function Treatment of Integu Head, Neck using Orthosis"
    },
    {
      "id": "4125dd75-a46e-4658-8aa8-9a0fecbebe96"
    },
    {
      "id": "7628788b-78ea-4664-a305-c4567868fdec",
      "name": "Dilate of 3 Cor Art with 2 Intralum Dev, Perc Endo Approach"
    },
    {
      "id": "60f7aa9a-37d3-42c6-81e5-03252220f164",
      "name": "Introduce Regional Anesth in Periph Nrv, Plexi, Perc"
    },
    {
      "id": "fb95ef40-2090-4d9c-ae0b-a7e36740df26"
    },
    {
      "id": "40297575-f54d-4d77-b6e2-e0b680996e51",
      "name": "Removal of Extralum Dev from Hepatobil Duct, Open Approach"
    },
    {
      "id": "2c610ebe-5682-4cc8-80f0-3788b74d4150"
    }
  ]
}
Exists, no beds
{
  "id": "626e8c70-dc93-44fb-a9d5-0b828a0a0c4b",
  "name": "Arrowood"
}
Does not exists:
ERROR:
  Code: InvalidArgument
  Message: id not found

